pkg/network/tracer/tracer.go

```diff
@@ -290,6 +286,46 @@ func NewTracer(config *config.Config) (*Tracer, error) {
 	return tr, nil
 }
 
+func initializePortBindingMaps(config *config.Config, m *manager.Manager) error {
+	if tcpPorts, err := network.ReadInitialState(config.ProcRoot, network.TCP, config.CollectIPv6Conns); err != nil {
+		return fmt.Errorf("failed to read initial TCP pid->port mapping: %s", err)
+	} else {
+		tcpPortMap, _, err := m.GetMap(string(probes.PortBindingsMap))
+		if err != nil {
+			return fmt.Errorf("failed to get TCP port binding map: %w", err)
+		}
+		for p := range tcpPorts {
+			log.Debugf("adding initial TCP port binding: netns: %d port: %d", p.Ino, p.Port)
+			pb := portBindingTuple{net_ns: C.__u32(p.Ino), port: C.__u16(p.Port)}
+			state := uint8(C.PORT_LISTENING)
+			err = tcpPortMap.Update(unsafe.Pointer(&pb), unsafe.Pointer(&state), ebpf.UpdateNoExist)
```

Comment:
Is there a race here between the kprobes updating this map and this code?

Reply:
This is happening before the kprobes are attached, so there is a small chance we would miss additional ports before the kprobes are attached. Do you have ideas on how to resolve this?

Reply:
Chatted offline. Added a ticket to improve this later.